src/core/undo/MovePageUndoAction.cpp

```diff
@@ -0,0 +1,28 @@
+#include "MovePageUndoAction.h"
+
+#include "util/i18n.h"              // for _
+
+MovePageUndoAction::MovePageUndoAction(const PageRef& page, size_t oldPos, size_t newPos):
+        UndoAction("MovePageUndoAction"),
+        oldPos(page, newPos, true),
+        newPos(page, oldPos, false) {
+    this->page = page;
+}
+
+MovePageUndoAction::~MovePageUndoAction() { this->page = nullptr; }
```

Comment:
Is there a hidden use to setting `page = nullptr` here?
```suggestion
MovePageUndoAction::~MovePageUndoAction() = default;
```